emission/analysis/userinput/matcher.py

```diff
@@ -38,57 +39,132 @@ def match_incoming_inputs(user_id, timerange):
-    if "trip_addition" not in confirmed_trip["data"] or \
-        confirmed_trip["data"]["trip_addition"] is None:
-        confirmed_trip["data"]["trip_addition"] = []
-    if "status" not in ui.data or ui.data.status == ecwtui.InputStatus.ACTIVE:
-        confirmed_trip["data"]["trip_addition"].append(ui)
-    elif ui.data.status == ecwtui.InputStatus.DELETED:
-        after_del_list = [ta for ta in confirmed_trip["data"]["trip_addition"] if ta["data"]["match_id"] != ui["data"]["match_id"]]
-        confirmed_trip["data"]["trip_addition"] = after_del_list
+        confirmed_obj["data"]["user_input"][input_name] = ui.data.label
+
+def handle_multi_non_deleted_match(confirmed_obj, ui):
+    logging.debug(f"handling user input {ui} for {confirmed_obj}")
+    if "additions" not in confirmed_obj["data"] or \
+        confirmed_obj["data"]["additions"] is None:
+        confirmed_obj["data"]["additions"] = []
+    if "status" not in ui.data or ui.data.status == ecwui.InputStatus.ACTIVE:
+        confirmed_obj["data"]["additions"].append(ui)
+    elif ui.data.status == ecwui.InputStatus.DELETED:
+        after_del_list = [ta for ta in confirmed_obj["data"]["additions"] if ta["data"]["match_id"] != ui["data"]["match_id"]]
+        confirmed_obj["data"]["additions"] = after_del_list
     else:
         # TODO: Decide whether to error or to warn here
         logging.warn("Invalid status found in user input %s, moving forward anyway" % ui)
 
+def create_composite_objects(user_id):
+    time_query = epq.get_time_range_for_composite_object_creation(user_id)
+    ts = esta.TimeSeries.get_time_series(user_id)
+    confirmedTrips = esda.get_entries(esda.CONFIRMED_TRIP_KEY, user_id, time_query=time_query)
+    logging.info("Creating composite trips from %d confirmed trips" % len(confirmedTrips))
+    if len(confirmedTrips) == 0:
+        logging.debug("len(confirmedTrips) == 0, early return")
+        return None
+
+    last_done_ts = None
+    for ct in confirmedTrips:
+        logging.info("End place type for trip is %s" %  type(ct.data.end_place))
+        composite_trip_dict = copy.copy(ct)
+        del composite_trip_dict["_id"]
+        composite_trip_dict["metadata"]["key"] = "analysis/composite_trip"
+
+        # confirmed_trip has an id for its corresponding confirmed_place
+        # for composite_trip, we want to get the actual confirmed_place object
+        confirmed_place_id = ct["data"]["confirmed_place"]
+        confirmed_place = esda.get_entry(esda.CONFIRMED_PLACE_KEY, confirmed_place_id)
+        composite_trip_dict["data"]["confirmed_place"] = confirmed_place
+
+        # retrieve locations for the trajectory of the trip
+        time_query = estt.TimeQuery("data.ts", ct["data"]["start_ts"], ct["data"]["end_ts"])
+        locations = esda.get_entries(esda.CLEANED_LOCATION_KEY, user_id, time_query=time_query)
+        max_entries = 100; # we will downsample to 100 locations
```

Comment:
later cleanup: you might want to make this be configurable. Not sure if it needs to be in the dynamic config, or can be in a file in `conf`

Reply:
I'll try to figure out dynamic config server-side